### ISSUE TITLE
iou_t vs iou_thres in training

### ISSUE THREAD

**Issue by niranjanravilevelup** [ultralytics/yolov5#6497]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,
 
     I am working with YoloV5 to train my custom dataset. While going through the train.py file, I was able to notice two declarations. 
1. 'iou_t': (0, 0.1, 0.7) --> Which is set to 0.2 in hyp.yaml
2. iou_thres in model.train() -->set to 0.6

I have two questions:

1. What significance does "iou_t" have in training? (Wasnt sure if it is being used anywhere)

2. Does setting iou_thres to 0.60, are we checking for overlap between ground truth and prediction boxes at 60% instead of 50% ? Cause even when it is set to 0.60, I could see result files having a column of mAP .5 and mAP 0.5:.95?

It would be very much helpful if anyone could guide me in this. 

Many thanks,
Niranjan

**Comment by glenn-jocher** [comment]
@niranjanravilevelup hyp['iou_t'] is the IoU threshold used for anchor-target assignment during training in loss.py

**Comment by niranjanravilevelup** [comment]
@glenn-jocher  Thank you for the clarification.

**Comment by blulightspecial** [comment]
@glenn-jocher, sorry to invoke you like this, but I don't think that `iou_t` is used there anymore. (_I actually don't think that it's used **anywhere** based on a search of the repo_)

It looks like the line that uses `iou_t` in loss.py is commented out and the `anchor_t` hyper parameter  is now responsible for helping with anchor-target assignment. 

https://github.com/ultralytics/yolov5/blob/47f265dde27d89cd1bd941bb9861a22334e2dcfa/utils/loss.py#L192-L194

Was the intention that users would comment out the upper `j=...` line and uncomment the lower `j=...` if they wanted to base anchor-target assignment on `iou_t` instead of `anchor_t`?



Thanks for any insight that you can provide. All of your 1000s of issue comments have been so helpful as I work with your library.

**Comment by glenn-jocher** [comment]
@blulightspecial yes, uncomment L193 to use IoU threshold rather than width-height threshold for target assignment.
